core/contract_test.go

```diff
@@ -75,46 +75,22 @@ func TestNewContract(t *testing.T) {
 		require.EqualError(t, err, core.ErrContractAlreadyDeployed.Error())
 	})
 
-	t.Run("instantiate previously deployed contract", func(t *testing.T) {
-		t.Run("contract with empty storage", func(t *testing.T) {
-			newContract, err := core.NewContract(addr, txn)
-			require.NoError(t, err)
-
-			root, err := newContract.Root()
-			require.NoError(t, err)
-			assert.Equal(t, &felt.Zero, root)
-		})
-		t.Run("contract with non-empty storage", func(t *testing.T) {
-			oldRoot, err := contract.Root()
-			require.NoError(t, err)
-
-			require.NoError(t, contract.UpdateStorage([]core.StorageDiff{{Key: addr, Value: classHash}}, NoopOnValueChanged))
-
-			newContract, err := core.NewContract(addr, txn)
-			require.NoError(t, err)
-
-			root, err := newContract.Root()
-			require.NoError(t, err)
-			assert.NotEqual(t, oldRoot, root)
-		})
-	})
-
```

Comment:
I don't see why these tests need to be removed.

Reply:
My reasoning was, since `Root` is not a member function of `Contract` anymore it doesnt really make sense to test its relation with a `NewContract` call

Reply:
That is a consequence of splitting write and read functions. I think we still need to maintain the same behaviour and the only way to check the updater has made the changes correctly is by calling the static functions as it is done in the tests below.

Reply:
That is already done at https://github.com/NethermindEth/juno/pull/1432/files#diff-a0246ffea9132d6506bb64b2651d4410c9942017252ee959741aed9268792178L179-L192 , tho?